feature_engine/timeseries/forecasting/base_forecast_transformers.py

```diff
@@ -165,6 +177,18 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
         if self.missing_values == "raise":
             self._check_na_and_inf(X)
 
+        if self.group_by_variables:
```

Comment:
Another thing that comes before this, in line 168, we are explicity forbidding duplicate values in the index, but if we allow groupby, we need to think what we do with this check

Reply:
@solegalli I think no duplicates will be in resultant dataframe, because we are creating the features for every group like the example below:
```
>>> import pandas as pd
    >>> from feature_engine.timeseries.forecasting import ExpandingWindowFeatures
    >>> X = pd.DataFrame(dict(date = ["2022-09-18",
    >>>                          "2022-09-19",
    >>>                          "2022-09-20",
    >>>                          "2022-09-21",
    >>>                          "2022-09-22",
    >>>                          "2022-09-18",
    >>>                          "2022-09-19",
    >>>                          "2022-09-20",
    >>>                          "2022-09-21",
    >>>                          "2022-09-22"],
    >>>                  x1 = [1,2,3,4,5, 3,5,6,8,11],
    >>>                  x2 = [6,7,8,9,10, 2,9,10,15,2],
    >>>                  x3=['a','a','a','a','a', 'b','b','b','b','b']
    >>>                ))
    >>> ewf = ExpandingWindowFeatures(group_by_variables='x3')
    >>> ewf.fit_transform(X)
```
the result is:
```
date  x1  x2 x3  x1_expanding_mean  x2_expanding_mean
    0  2022-09-18   1   6  a                NaN                NaN
    1  2022-09-19   2   7  a           1.000000                6.0
    2  2022-09-20   3   8  a           1.500000                6.5
    3  2022-09-21   4   9  a           2.000000                7.0
    4  2022-09-22   5  10  a           2.500000                7.5
    5  2022-09-18   3   2  b                NaN                NaN
    6  2022-09-19   5   9  b           3.000000                2.0
    7  2022-09-20   6  10  b           4.000000                5.5
    8  2022-09-21   8  15  b           4.666667                7.0
    9  2022-09-22  11   2  b           5.500000                9.0
```